include/pybind11/common.h

```diff
@@ -30,7 +30,7 @@
 #  define PYBIND11_NOINLINE __attribute__ ((noinline))
 #endif
 
-#if __cplusplus > 201103L
+#if PYBIND11_CPP14
```

Comment:
Hmm -- that doesn't seem right. ``PYBIND11_CPP14`` isn't defined until later below.